### PR TITLE
added timeout parameter in download_sample_data()

### PR DESCRIPTION
**Pull request by sanskar-modi**
Default timeout for download_sample_data() is equal to `astropy.utils.data.Conf.remote_timeout` i.e 3sec. Believe me i'm on good connection and still it gives me timeout error everytime and i had to download sample data using `overwrite=False`.
Adding a `timeout` parameter will be helpful for users on slow connections.

### CODE REVIEW

**Comment by Cadair** [comment]
Thanks!

Could you add an entry to the changelog please? (put `[ci skip]` in the commit message somewhere to stop the CI running).

**Comment by ayshih** [comment]
:+1: after changelog

File: sunpy/data/_sample.py [hunk 1]
```diff
@@ -66,6 +66,9 @@ def download_sample_data(progress=True, overwrite=True):
         Show a progress bar during download
     overwrite: bool
         If exist overwrites the downloaded sample data.
+    timeout: float
+        The timeout in seconds.
+        Default timeout is `astropy.utils.data.Conf.remote_timeout`.
```
Comment (ayshih) [inline sunpy/data/_sample.py]: Per discussion, change this line to say "If None, use the timeout specified in..."

**Comment by Cadair** [comment]
:+1:

File: sunpy/data/_sample.py (hunk 1 above)
Comment (Cadair) [inline sunpy/data/_sample.py]: fixed in merge

**Comment by sanskar-modi** [comment]
I think we should add try, except block around this
`f = download_file(os.path.join(base_url, full_file_name), timeout=timeout)` followed by a info message in except block about using either setting a `timeout` or `overwrite=False` as most of people downloading sample data will be new comers and they might not be aware of this thing.
Currently download_sample_data() doesn't handle timeout well, giving a traceback.

**Comment by Cadair** [comment]
Makes sense, feel free to open another PR.

**Pull request merged**